autodoc/data/admin-api.lua

```diff
@@ -390,6 +391,33 @@ return {
         },
       },
     },
+    metadata = {
+      title = [[ Metadata ]],
+      description = [[ Developer-supplied data. ]],
```

Comment:
This probably needs at least a paragraph-long, explaining that this is arbitrary free-form data, etc.

Reply:
There is an implicit assumption that almost everything on the admin api is part of the compatibility promise.

If this is going to be free form data, it would help us in future if we clearly document the compatibility promise here for Open source.

Reply:
I know I asked for this feature, but I get cold feet when we save user-generated, user-visible, free-form data, especially without a plan for localization. Plugin category (Authentication, Logging, etc) and possibly icon URL seem less problematic, but this begs the question whether all of this should go in an external registry as has been suggested elsewhere.